appinventor/blocklyeditor/src/replmgr.js

```diff
@@ -348,10 +349,13 @@ Blockly.ReplMgr.putYail = (function() {
                                 var hunk = json[i];
                                 var candidate = hunk['candidate'];
                                 offer = hunk['offer'];
-                                if (candidate && haveoffer) {
+                                if (candidate && haveoffer && connectionstate != "none") {
                                     var nonce = hunk['nonce'];
                                     if (!seennonce[nonce]) {
                                         seennonce[nonce] = true;
+                                        console.log("addIceCandidate: signaliingstate = " + webrtcpeer.signalingState +
+                                                    " iceConnectionState = " + webrtcpeer.iceConnectionState);
+                                        console.log("addIceCandidate: candidate = " + JSON.stringify(candidate));
                                         webrtcpeer.addIceCandidate(candidate);
```

Comment:
Yeah, well I actually tried this, but the GWT Compiler wouldn't have it. Without the catch the errors are thrown and result in an error. I know because while debugging this I managed to get things into a state where addIceCandidate would throw an exception.

Reply:
`[ERROR] Line 111698: missing name after . operator
                     webrtcpeer.addIceCandidate(candidate).catch(function(e) {
`

Reply:
Right. They'll treat catch as the catch keyword even though it's not valid in this context. We might be able to use `["catch"]` since global objects won't be minified anyway. I recall that last time I ran into this issue I hacked GWT's parser to deal with it, but using the bracket notation here is probably the better option.

Reply:
OK. I used this approach in the code I just pushed. It works (I actually purposely created an error to make sure the code is really invoked, it is). I also verified this works when we use the plover compiler's "SIMPLE" optimization feature, which we do for production.

I also slightly refactored error handling in the webrtc code.